pkg/storage/etcd3/etcdwatcher/event.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcdwatcher
+
+import "github.com/coreos/etcd/storage/storagepb"
+
+type event struct {
+	key       string
+	value     []byte
+	rev       int64
+	isDeleted bool
+	isCreated bool
+}
+
+func parseEtcdKV(kv *storagepb.KeyValue) *event {
+	return &event{
+		key:       string(kv.Key),
+		value:     kv.Value,
+		rev:       kv.ModRevision,
+		isDeleted: false,
+		isCreated: kv.ModRevision == kv.CreateRevision,
+	}
+}
+
+func parseEtcdEvent(etcdEvent *storagepb.Event) *event {
+	return &event{
+		key:       string(etcdEvent.Kv.Key),
+		value:     etcdEvent.Kv.Value,
+		rev:       etcdEvent.Kv.ModRevision,
+		isDeleted: etcdEvent.Type == storagepb.DELETE || etcdEvent.Type == storagepb.EXPIRE,
+		isCreated: etcdEvent.Type == storagepb.PUT && etcdEvent.Kv.ModRevision == etcdEvent.Kv.CreateRevision,
```

Comment:
@magicwang-cn - that's not my question.
I'm asking whether there is situation where "ModRev == CreateRev" and "isCreated" should be false